server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -2157,6 +2240,14 @@ && isNotUnitVector(squaredMagnitude)) {
             }
 
             int adjustedK = k;
+            // By default utilize the quantized oversample is configured
+            // allow the user provided at query time overwrite
+            Float oversample = indexOptions instanceof QuantizedIndexOptions quantizedIndexOptions
+                ? quantizedIndexOptions.rescoreVector != null ? quantizedIndexOptions.rescoreVector.oversample() : null
+                : null;
+            if (queryOversample != null) {
+                oversample = queryOversample;
+            }
```

Comment:
Nit - I find hard to follow multiple ternary operators

```suggestion
            Float oversample = null;
            if (queryOversample != null) {
                oversample = queryOversample;
            } else if (queryOversample indexOptions instanceof QuantizedIndexOptions quantizedIndexOptions) {
                oversample = quantizedIndexOptions.rescoreVector != null ? quantizedIndexOptions.rescoreVector.oversample() : null
            }
```

Reply:
I will refactor